.github/workflows/dockerd.yml

```diff
@@ -0,0 +1,139 @@
+name: dockerd
+
+on:
+  # TODO: add event to build on command in PR (e.g., /test-dockerd)
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Docker version'
+        required: true
+        default: '20.10.13'
+
+env:
+  REPO_SLUG_ORIGIN: "moby/buildkit:latest"
+  CACHE_GHA_SCOPE_IT: "integration-tests"
+  CACHE_GHA_SCOPE_BINARIES: "binaries"
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Check version
+        run: |
+          version=${{ github.event.inputs.version }}
+          if [ -z "$version" ]; then
+            version=20.10.13
+          fi
+          echo "DOCKER_VERSION=$version" >> $GITHUB_ENV
+      -
+        name: Check build
+        uses: actions/github-script@v6
+        id: build
+        with:
+          result-encoding: string
+          script: |
+            try {
+              new URL("${{ env.DOCKER_VERSION }}");
+            } catch (e) {
+              return false;
+            }
+            return true;
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: image=${{ env.REPO_SLUG_ORIGIN }}
+          buildkitd-flags: --debug
+      -
+        name: Build
+        if: steps.build.outputs.result == 'true'
+        uses: docker/build-push-action@v2
+        with:
+          context: ${{ env.DOCKER_VERSION }}
+          target: binary
+          outputs: /tmp/moby
+      -
+        # FIXME: remove symlink and rename bin. should be fixed upstream on moby.
+        name: Rename binary
+        if: steps.build.outputs.result == 'true'
+        run: |
+          rm /tmp/moby/binary-daemon/dockerd
+          mv /tmp/moby/binary-daemon/dockerd-dev /tmp/moby/binary-daemon/dockerd
+      -
+        name: Download
+        if: steps.build.outputs.result != 'true'
+        run: |
+          mkdir -p /tmp/moby/binary-daemon
+          cd /tmp/moby/binary-daemon
+          wget -qO- "https://download.docker.com/linux/static/stable/x86_64/docker-${{ env.DOCKER_VERSION }}.tgz" | tar xvz --strip 1
+      -
+        name: Upload dockerd
```

Comment:
Could we remove this artifact after workflow completes?

Reply:
hum does not seem possible atm but I can merge both jobs so we don't rely on this.